### PR TITLE
Example app update

### PR DESCRIPTION
**Pull request by Hung6129**

### CODE REVIEW

File: example/lib/src/views/control_module/node.dart [hunk 1]
```diff
@@ -47,7 +49,7 @@ class _NodeState extends State<Node> {
     if (!isLoading) {
       body = ListView(
         children: [
-          Text('Node $nodeAddress'),
+          Text(nodeName ?? 'Node $nodeAddress'),
           Text(widget.node.uuid),
```
Comment (ghost) [inline example/lib/src/views/control_module/node.dart]: Don't hide unicast adr
```
Text('${nodeName ?? 'Node'} $nodeAddress' ),
```

File: example/lib/src/views/home/home.dart [hunk 2]
```diff
@@ -43,6 +45,8 @@ class _HomeState extends State<Home> {
         _meshNetwork = event;
       });
     });
+
+    //
```
Comment (ghost) [inline example/lib/src/views/home/home.dart]: Remove empty lines

File: example/lib/src/views/home/home.dart [hunk 3]
```diff
@@ -149,6 +153,16 @@ class MeshNetworkDatabaseWidget extends StatelessWidget {
...
+                  Uint8List data = Uint8List.fromList(meshNetworkJson!.codeUnits);
+                  MimeType type = MimeType.OTHER;
+                  String path = await FileSaver.instance.saveAs("nRF_MeshNetwork", data, "json", type);
+                  debugPrint(path);
+                  debugPrint("Downloading");
+                  try {
+                    debugPrint("Download Completed.");
+                  } catch (e) {
+                    debugPrint("Download Failed.\n\n$e");
+                  }
```
Comment (ghost) [inline example/lib/src/views/home/home.dart]:
```suggestion
                  debugPrint("Downloading");
                  try {
                    Uint8List data = Uint8List.fromList(meshNetworkJson!.codeUnits);
                  MimeType type = MimeType.OTHER;
                  String path = await FileSaver.instance.saveAs("nRF_MeshNetwork", data, "json", type);
                    debugPrint("Download Completed.");
                  debugPrint(path);
                  } catch (e) {
                    debugPrint("Download Failed.\n\n$e");
                  }
```

File: example/lib/src/views/scan_and_provisionning/scan_and_provisioning.dart [hunk 4]
```diff
@@ -91,8 +91,10 @@ class _ScanningAndProvisioningState extends State<ScanningAndProvisioning> {
 
       if (Platform.isAndroid) {
         deviceUUID = _serviceData[device.id].toString();
+        debugPrint('start provisioning on android with device id: $deviceUUID');
```
Comment (ghost) [inline example/lib/src/views/scan_and_provisionning/scan_and_provisioning.dart]: Remove log

File: example/lib/src/views/scan_and_provisionning/scan_and_provisioning.dart [hunk 5]
```diff
@@ -91,8 +91,10 @@ class _ScanningAndProvisioningState extends State<ScanningAndProvisioning> {
...
       } else if (Platform.isIOS) {
         deviceUUID = device.id.toString();
+        debugPrint('start provisioning on iOS with device id: $deviceUUID');
```
Comment (ghost) [inline example/lib/src/views/scan_and_provisionning/scan_and_provisioning.dart]: Remove log

File: lib/src/utils/provisioning.dart [hunk 6]
```diff
@@ -262,6 +262,7 @@ Future<ProvisionedMeshNode> _provisioning(
     await bleMeshManager.refreshDeviceCache();
     await bleMeshManager.disconnect();
     _cancelProvisioningCallbackSubscription(bleMeshManager);
+    provisionedMeshNode.nodeName = deviceToProvision.name;
```
Comment (ghost) [inline lib/src/utils/provisioning.dart]: I feel like it should not be done here or not like this. Can we try to do it earlier? For example, when the UnprovisionedMeshNode object is created ? Or maybe expose the parameter name in provisioning method..🤔

File: example/lib/src/views/control_module/model.dart [hunk 7]
```diff
@@ -8,8 +8,20 @@ class Model extends StatelessWidget {
...
   Widget build(BuildContext context) {
-    return Wrap(
-      children: <Widget>[Text(' ${model.modelId} '), appKeyBindIcon(), const Text(', ')],
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.start,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        Row(
+          children: [
+            Text("ModelName:${getSigMeshName("0x${model.modelId.toRadixString(16).padLeft(4, '0').toUpperCase()}")}"),
```
Comment (ghost) [inline example/lib/src/views/control_module/model.dart:17]: Extract the hex value and use it here and in the other Text widget

**Review by ghost — CHANGES_REQUESTED** [review]
Hello @Hung6129 ,
Thanks for the PR 👍
A few changes before we merge..

File: lib/src/utils/provisioning.dart (hunk 6 above)
Comment (Hung6129) [reply lib/src/utils/provisioning.dart]: Oh 
I think do it earlier it good too
It just i do not know a efficient way to do this 😢 
If in the provisioning method where should i call it tho and same with UnprovisionedMeshNode object? 
hope you can show me where is the best way to call and expose it
thanks

File: example/lib/src/views/control_module/node.dart (hunk 1 above)
Comment (Hung6129) [reply example/lib/src/views/control_module/node.dart]: My bad 😄

**Comment by Hung6129** [comment]
Oh, about the node name
I think do it earlier it good too
It just i do not know a efficient way to do this 😢
If in the provisioning method where should i call it tho and same with UnprovisionedMeshNode object?
hope you can show me where is the best way to call and expose it
thanks

**Comment by ghost** [comment]
Hi @Hung6129 ,

Changing the name ealier probably implies to edit some native code and I couldn't find it easily on GitHub to guide you. Maybe we can split your changes : revert the changes on the plugin and only keep changes in example app. If you follow the comments I added then we can merge the example app update and work on choosing name upon provisioning in a separate PR ?

**Comment by Hung6129** [comment]
Yeah sure !! 
Can you tell me when we can start?
And what can i do to make it easier for you?

Btw it was mine first time to create a PR on a repo, hope u can skip out on the mistake 😟 

Thanks you

**Comment by ghost** [comment]
You didn't make any mistakes. Thanks for your contribution 😃 

Make a commit on your branch that applies my suggestions and that removes the name change at the end of provisioning. So we can merge the example app update.
Then open a new issue for the feature request about having the newly provisioned node named after the given DiscoveredDevice.

File: example/lib/src/views/home/home.dart [hunk 8]
```diff
@@ -53,6 +57,8 @@ class _HomeState extends State<Home> {
     super.dispose();
   }
 
+
```
Comment (ghost) [inline example/lib/src/views/home/home.dart]: Remove empty lines

**Comment by ghost** [comment]
Thx for the removal and for other updates in the issues 👍
There are still some needed changes before we can merge 🚀

**Review by ghost — CHANGES_REQUESTED** [review]
Hello @Hung6129 

Would you have some time to apply my suggested changes ?
So we can merge your PR ;)

File: example/lib/src/views/home/home.dart (hunk 8 above)
Comment (Hung6129) [reply example/lib/src/views/home/home.dart]: ok sure

Comment (Hung6129) [reply example/lib/src/views/home/home.dart]: ok i'm done now i create another PR ?

**Review by Hung6129 — COMMENTED** [review]
Resolve the changes

File: pubspec.lock [hunk 9]
```diff
@@ -433,107 +476,122 @@ packages:
...
       name: yaml
-      url: "https://pub.dartlang.org"
+      sha256: "23812a9b125b48d4007117254bca50abb6c712352927eece9e155207b1db2370"
+      url: "https://pub.dev"
     source: hosted
-    version: "3.1.0"
+    version: "3.1.1"
 sdks:
-  dart: ">=2.17.0 <3.0.0"
+  dart: ">=2.19.0 <4.0.0"
```
Comment (ghost) [inline pubspec.lock]: I'd rather not merge such change as of now. Dart 2.17 is fine, isn't it ?

File: example/pubspec.lock [hunk 10]
```diff
@@ -469,142 +570,170 @@ packages:
...
       name: yaml
-      url: "https://pub.dartlang.org"
+      sha256: "23812a9b125b48d4007117254bca50abb6c712352927eece9e155207b1db2370"
+      url: "https://pub.dev"
     source: hosted
-    version: "3.1.0"
+    version: "3.1.1"
 sdks:
-  dart: ">=2.17.0 <3.0.0"
+  dart: ">=2.19.0 <4.0.0"
```
Comment (ghost) [inline example/pubspec.lock]: See comment on other pubspec.lock

**Review by ghost — CHANGES_REQUESTED** [review]
Thanks 😊 LGTM except the pubspec.lock files update as it implies higher dart sdk constraint

**Pull request closed without merging**